CMakeLists.txt

```diff
@@ -10,10 +10,10 @@ option(BUILD_CURL_TRANSPORT "Build internal http transport implementation with C
 option(UNIT_TESTING "Build unit test projects" OFF)
 option(UNIT_TESTING_MOCK_ENABLED "wrap PAL functions with mock implementation for tests" OFF)
 option(BUILD_PAHO_TRANSPORT "Build IoT Samples with Paho MQTT support" OFF)
-option(BUILD_PRECONDITIONS "Build SDK with preconditions enabled" ON)
+option(BUILD_WITH_PRECONDITIONS "Build SDK with preconditions enabled" ON)
```

Comment:
See the other option names.  In cmake options world, it is uncommon to use `WITH`.
I would even change `UNIT_TESTING` to `BUILD_TESTING`  as we have in C++.

So, before merging this, you might to make sure `WITH` is going to be liked by users, since you are making option name longer

Reply:
@ahsonkhan WITH is not used on any options.  We should use it here either.

I like dropping the BUILD portion:
BUILD_PRECONDITIONS -> PRECONDITONS

Then as an option its clear:
PRECONDITIONS=ON or PRECONDITIONS=OFF

Reply:
I also don't think removing `BUILD` is a good idea.
My understanding is that any cmake option that starts with `BUILD...` is for generating/compiling an entire lib/module/exe.   
Whereas any other option without `BUILD...` prefix is for a setting (like PRECONDITIONS).

So, I am fine with renaming to `PRECONDITIONS`, but curl_tranport and paho should still have the `BUILD_` prefix so users know it is a removable lib/exe/module.

Here is our expect anyway, @barcharcraz ,  for advicing

Reply:
The names are more or less arbitrary, although some, like BUILD_TESTING are used inside modules that are shipped with cmake. They don't even have to be ALL_CAPS.

One thing to keep in mind is the cmake-gui application will group options that have a common prefix. The important thing is consistency and documenting that consistency.

There's no real de-facto standard for this naming either so it really is up to individual projects.

Reply:
Is that a common convention you have seen followed elsewhere? And is there strong consensus that we should be following that convention?

The cmake options seem quite open ended:
https://cmake.org/cmake/help/v3.6/manual/cmake.1.html

Reply:
@vhvb1989 - given Charlie's thoughts, are you fine with this change as is?

Reply:
yes, this is fine.
I would really like to make a difference from Options that adds/builds something v/s those options that set up configuration.  So when listing options, the two groups can be seen separate.
But, it is not really important, and the Prefix, based on Charlie, won't matter at all :)